app/javascript/packages/address-search/index.tsx

```diff
@@ -84,26 +79,30 @@ export const transformKeys = (location: object, predicate: (key: string) => stri
     {},
   );
 
-const requestUspsLocations = async (address: LocationQuery): Promise<FormattedLocation[]> => {
-  const response = await request<PostOffice[]>(LOCATIONS_URL, {
+const requestUspsLocations = async (
+  locationsUrl: string,
+  address: LocationQuery,
+): Promise<FormattedLocation[]> => {
+  const response = await request<PostOffice[]>(locationsUrl, {
     method: 'post',
     json: { address: transformKeys(address, snakeCase) },
   });
 
   return formatLocations(response);
 };
 
-export const ADDRESS_SEARCH_URL = new URL('/api/addresses', window.location.href).toString();
-
-function requestAddressCandidates(unvalidatedAddressInput: string): Promise<Location[]> {
-  return request<Location[]>(ADDRESS_SEARCH_URL, {
+function requestAddressCandidates(
+  unvalidatedAddressInput: string,
+  addressSearchUrl: string,
+): Promise<Location[]> {
+  return request<Location[]>(addressSearchUrl, {
     method: 'POST',
     headers: { 'Content-Type': 'application/json' },
     json: { address: unvalidatedAddressInput },
   });
 }
 
-function useUspsLocations() {
+function useUspsLocations(locationsUrl: string, addressSearchUrl: string) {
```

Comment:
I'd make these keyword arguments instead of ordered ones, because flipping the order accidentally could cause problems
```suggestion
function useUspsLocations({ locationsUrl, addressSearchUrl} : { locationsUrl: string, addressSearchUrl: string }) {
```